dbms/src/Functions/FunctionsRegexp.h

```diff
@@ -973,15 +1126,380 @@ class FunctionStringRegexp : public FunctionStringRegexpBase
+                auto nullmap_col = ColumnUInt8::create();
+                typename ColumnUInt8::Container & null_map = nullmap_col->getData();
+                null_map.resize(col_size);
+
+                for (size_t i = 0; i < col_size; ++i)
+                {
+                    if (expr_param.isNullAt(i) || pos_param.isNullAt(i) || occur_param.isNullAt(i) || ret_op_param.isNullAt(i))
+                    {
+                        null_map[i] = 1;
+                        continue;
+                    }
+                    null_map[i] = 0;
+                    expr_param.getStringRef(i, expr_ref);
+                    GET_POS_VALUE(i)
+                    GET_OCCUR_VALUE(i)
+                    GET_RET_OP_VALUE(i)
+                    vec_res[i] = regexp->instr(expr_ref.data, expr_ref.size, pos, occur, ret_op);
+                }
+                res_arg.column = ColumnNullable::create(std::move(col_res), std::move(nullmap_col));
+            }
+            else
+            {
+                // Process pure vector columns with memorized regexp.
+                // columns are impossible to be a nullable column here.
+
+                for (size_t i = 0; i < col_size; ++i)
+                {
+                    expr_param.getStringRef(i, expr_ref);
+                    GET_POS_VALUE(i)
+                    GET_OCCUR_VALUE(i)
+                    GET_RET_OP_VALUE(i)
+                    vec_res[i] = regexp->instr(expr_ref.data, expr_ref.size, pos, occur, ret_op);
+                }
+                res_arg.column = std::move(col_res);
+            }
+        }
+        else
+        {
+            // Codes in this if branch execute instr without memorized regexp
+
+            if constexpr (has_nullable_col)
+            {
+                // Process nullable columns without memorized regexp
+                auto nullmap_col = ColumnUInt8::create();
+                typename ColumnUInt8::Container & null_map = nullmap_col->getData();
+                null_map.resize(col_size);
+
+                for (size_t i = 0; i < col_size; ++i)
+                {
+                    if (expr_param.isNullAt(i) || pat_param.isNullAt(i) || pos_param.isNullAt(i) || occur_param.isNullAt(i) || ret_op_param.isNullAt(i) || match_type_param.isNullAt(i))
+                    {
+                        null_map[i] = 1;
+                        continue;
+                    }
+                    null_map[i] = 0;
+                    expr_param.getStringRef(i, expr_ref);
+                    pat = pat_param.getString(i);
+                    if (unlikely(pat.empty()))
+                        throw Exception(EMPTY_PAT_ERR_MSG);
+                    GET_POS_VALUE(i)
+                    GET_OCCUR_VALUE(i)
+                    GET_RET_OP_VALUE(i)
+                    match_type = match_type_param.getString(i);
+                    auto regexp = createRegexpWithMatchType(pat, match_type, collator);
```

Comment:
Seems we can not put the regexp into to pool if regexp can not be memoryized, otherwise server may get oom since the ObjectPool actually has no evict mechanism, the regexp put into the pool will be hold forever.

Reply:
> Seems we can not put the regexp into to pool if regexp can not be memoryized, otherwise server may get oom since the ObjectPool actually has no evict mechanism, the regexp put into the pool will be hold forever.

Done